priv/user_linux.go

```diff
@@ -151,3 +150,20 @@ func setresuid(ruid, euid, suid int) error {
 	}
 	return nil
 }
+
+func setEnvironmentForUser(user *user.User) error {
+	if err := os.Setenv("HOME", user.HomeDir); err != nil {
+		return err
+	}
+	if err := os.Setenv("USER", user.Name); err != nil {
+		return err
+	}
+	if _, ok := os.LookupEnv("DOCKER_CONFIG"); ok {
+		return nil
+	}
+	// ggcr sets default docker config during init, fix for user
+	if err := os.Setenv("DOCKER_CONFIG", filepath.Join(user.HomeDir, ".docker")); err != nil {
+		return err
+	}
```

Comment:
I think we should remove this code, `DOCKER_CONFIG` is only relevant to analyzer/exporter and this function only runs with `creator`, we wouldn't want analyzer/exporter to lookup docker config differently as part of creator vs. as a stand alone phase. We originally put this in to be extra cautious, but I think it was the wrong decision for following reasons:

1. This code only runs when the lifecycle is executes as root
2. `pack` doesn't use the docker config to pass creds to the lifecycle
3. other platforms that do use the docker config don't run these phases as root
4. other platforms that do use the docker config (`tekton`, `kpack`) explicitly set `HOME` already, in order to provide docker config generated from secrets, this logic would probably break them rather than help them if they did run as root